src/references/referenceService.ts

```diff
@@ -0,0 +1,230 @@
+    this.inputText = inputBody;
+    const output = this.refFormatter(this.inputText);
+    return {
+      formattedText: output.text,
+      references: output.referenceList,
+    };
+  }
+
+  refFormatter(inputText: string) {
+    let remainingText: string = "";
+    let instanceCounter = 1;
+
+    // Track where each reference occurs
+    const refInstance = this.identifyReferences(inputText);
+    for (const instance of refInstance) {
+      const newRefInstance: IReferenceInstance = {
+        nameYear: instance[0],
+        index: instance.index!,
+      };
+      this.referenceInstanceLibrary.set(instanceCounter, newRefInstance);
+      instanceCounter++;
+    }
+
+    // Build a collection of reference authors, indexes of their references, and length of each reference
+    // Checks to see if there's a reference directly after the one currently being focused on
+    const rawRefs = this.identifyReferences(inputText);
+    for (const rawRef of rawRefs) {
+      if (this.recordExists(rawRef)) {
+        const existingRef = this.referenceLibrary.get(rawRef[0]);
+        const existingLocations = existingRef.locations;
+        const newLocation = rawRef.index;
+        existingLocations.push(newLocation);
+        this.arrayOfRefsForFormatting.push(existingRef.referenceId);
+      } else if (!this.recordExists(rawRef)) {
+        const newRef: IReference = {
+          referenceId: this.referenceLibrary.size + 1,
+          locations: [rawRef.index!],
+          referenceLength: rawRef[0].length,
+        };
+        this.referenceLibrary.set(rawRef[0], newRef);
+        this.arrayOfRefsForFormatting.push(newRef.referenceId);
+      }
+      remainingText = inputText.slice(rawRef.index! + rawRef[0].length + 1);
+
+      // Check text immediately after the current reference. If nothing adjacent, format existing collection of references.
+      this.identifyAdjacentReferences(remainingText);
+      this.refInstanceCount++;
+    }
+
+    const referencesLeftToFormat =
+      this.arrayOfRefsForFormatting.length > 0 ? true : false;
+
+    if (referencesLeftToFormat) {
+      this.concatReferencesToFinalOutput();
+    }
+    let referenceList = this.createReferenceList(this.referenceLibrary);
+    return { text: this.finalText, referenceList: referenceList };
+  }
+
+  // Helper Functions
+  identifyReferences(text: string) {
+    return text.matchAll(this.REFERENCE_IDENTIFICATION_REGEX);
+  }
+
+  recordExists(rawRef: RegExpMatchArray) {
+    const reference = rawRef[0];
+    if (this.referenceLibrary.get(reference) === undefined) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  identifyAdjacentReferences(string: string) {
+    if (!string) {
+      return;
+    }
+    if (!this.isOpenSquareBracket(string[0])) {
+      this.concatReferencesToFinalOutput();
+    }
+  }
+
+  concatReferencesToFinalOutput() {
+    let startOfReferenceIndex: number = 0;
+    let startOfSectionIndex: number = 0;
```

Comment:
```suggestion
    let startOfReferenceIndex = 0;
    let startOfSectionIndex = 0;
```

If you instantiate a variable with a value, there's no need to give it a type _unless_ you're giving it a union type, e.g., 

```
let x: string | number = 'hi';
x = 1; // still valid
```

Reply:
Ah, thought assigning types was always best practice!